features/measurements/impl/src/test/java/org/opennms/netmgt/measurements/impl/JEXLExpressionEngineEnhancedTest.java

```
@@ -0,0 +1,307 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.measurements.impl;
+
+import static org.junit.Assert.*;
+
+import java.util.Map;
+
+import org.apache.commons.jexl2.JexlContext;
+import org.junit.Test;
+import org.opennms.netmgt.measurements.api.ExpressionEngine;
+import org.opennms.netmgt.measurements.api.exceptions.ExpressionException;
+import org.opennms.netmgt.measurements.api.FetchResults;
+import org.opennms.netmgt.measurements.model.Expression;
+import org.opennms.netmgt.measurements.model.QueryRequest;
+import org.opennms.netmgt.measurements.model.Source;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+public class JEXLExpressionEngineEnhancedTest {
+
+	private final ExpressionEngine jexlExpressionEngine = new JEXLExpressionEngine();
+
+	@Test
+	public void checkSamplefn5NaN(){
+		final String expression = "fn:arrayNaN(\"x\", 5)";
+		boolean success = true;
+
+		StringBuffer sb = new StringBuffer("");
+		try{
+
+			double[] result = performExpression(expression);
+
+			for (double value: result){
+				sb.append(value+",");
+			}
+
+			success = 
+					Double.isNaN(result[0]) &&
+					Double.isNaN(result[1]) &&
+					Double.isNaN(result[2]) &&
+					Double.isNaN(result[3]) &&
+					Double.isNaN(result[4]) &&
+					(result[5] == 5) &&
+					(result[6] == 6);
+
+		} catch (Exception e){
```

Why not just re-throw the exception instead of catching it here? The test will fail the and the stack will be printed automatically.